processing/src/main/java/org/apache/carbondata/processing/loading/sort/SortStepRowHandler.java

```diff
@@ -359,9 +433,14 @@ public void writeRawRowAsIntermediateSortTempRowToOutputStream(Object[] row,
 
     // write no-dict & sort
     for (int idx = 0; idx < this.noDictSortDimCnt; idx++) {
-      byte[] bytes = (byte[]) row[this.noDictSortDimIdx[idx]];
-      outputStream.writeShort(bytes.length);
-      outputStream.write(bytes);
+      if (DataTypeUtil.isPrimitiveColumn(noDicSortDataTypes[idx])) {
```

Comment:
I can see that at multiple places for every row DataTypeUtil.isPrimitiveColumn is getting used. Please check the load performance impact of this